File: Intf/Model/FornecedorModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Intf
{
	public class FornecedorModel : IModel
	{
		public Guid Id { get; set; }
		public DateTime? CreatedAt { get; set; }
		public DateTime? UpdatedAt { get; set; }
		public string? CreatedBy { get; set; }
		public string? UpdatedBy { get; set; }

		public string? Cnpj { get; set; }
		public string? Cpf { get; set; }
		public string? Rg { get; set; }
		public FornecedorTypeEnum Type { get; set; }
		public DateTime? BirthDate { get; set; }
		public string Name { get; set; } = string.Empty;
		public int Cep { get; set; }
		public string Email { get; set; } = string.Empty;

		public ICollection<EmpresaFornecedorModel> EmpresaFornecedores { get; set; } = new List<EmpresaFornecedorModel>();
	}
}

File: frontend/src/app/models/fornecedor.model.ts
export enum FornecedorType {
  PessoaFisica = 0,
  PessoaJuridica = 1
}

export interface Fornecedor {
  id: string;
  cnpj?: string | null;
  cpf?: string | null;
  rg?: string | null;
  type: FornecedorType;
  birthDate?: string | null;
  name: string;
  cep: number;
  email: string;
}
